docs/auth.md

```diff
@@ -0,0 +1,259 @@
+   trusted to masquerade as any other identity.  When running behind the
+   API aggregator, this *must* be the same CA used to sign the
+   aggregator's proxy client certificate.  When not running with an
+   aggregator (e.g. pre-Kubernetes-1.7, without a separate aggregator
+   pod), this simply needs to exist.
+
+### Quickstart
+
+**Note**: options on the service catalog API server take paths to files.
+Options passed to the Helm chart take the base64-encoded contents of the
+files.
+
+The bare minimum for running a service catalog API server requires
+
+1. Ensure that you have a ConfigMap in the `kube-system` namespace named
+   `extensions-apiserver-authentication`.  If not, simply follow the "if
+   not, ..." clauses in steps 2 and 3.
+
+   ```shell
+   $ kubectl get configmap extension-apiserver-authentication --namespace kube-system -o yaml
+   ```
+
+2. Ensure that there is a `client-ca-file` key in that ConfigMap.  If not,
+   you'll need a client CA file (see [generating
+   certificates](#generating-certificates) below).  This file can be
+   passed to the service catalog API server using `--client-ca-file`.
+
+3. Ensure that there is a `requestheader-client-ca-file` key in that
+   ConfigMap. If not, you'll need a RequestHeader client CA file (see
+   [generating certificates](#generating-certificates) below).  This file
+   can be passed to the service catalog API server using
+   `--requestheader-client-ca-file`, or the
+   `apiserver.tls.requestHeaderCA` Helm chart option.
+
+4. If not running on the cluster, ensure that you pass in a Kubeconfig
+   file to authenticate with the main cluster using the
+   `--authentication-kubeconfig` and `--authorization-kubeconfig` options
+   on the service catalog API server.
+
+### Generating certificates
+
+The Kubernetes documentation has a [detailed
+section](https://kubernetes.io/docs/admin/authentication/#creating-certificates)
+on how to create certificates several different ways.  For convinience,
+we'll reproduce the basics using the `openssl` and `cfssl` commands below
+(you can install `cfssl` using `go get -u
+github.com/cloudflare/cfssl/cmd/...`).
+
+In the common case, all three CA certificates referenced above already
+exist as part of the main Kubernetes cluster setup.
+
+In case you need to generate any of the CA certificate pairs mentioned
+above yourself, you can do so using the following command (see below for
+appropriate values of `$PURPOSE`):
+
+```shell
+export PURPOSE=<purpose>
+openssl req -x509 -sha256 -new -nodes -days 365 -newkey rsa:2048 -keyout ${PURPOSE}-ca.key -out ${PURPOSE}-ca.crt -subj "/CN=ca"
+echo '{"signing":{"default":{"expiry":"43800h","usages":["signing","key encipherment","'${PURPOSE}'"]}}}' > "${PURPOSE}-ca-config.json"
+```
+
+This generates a certificate and private key for the CA, as well as
+a signing configuration used by `cfssl` below. `$PURPOSE` should be set to
+one of `serving`/`server`, `client`, or `requestheader-client`, as
```

Comment:
I can verify the CA part plus the base64 part in two paragraphs works for getting past #737

The serving certificate stuff requires some testing. We should probably try and get some e2e on it.